### PR TITLE
f-checkout@0.45.0 - Only call the Checkout `GET` endpoint when the user is logged in

### PR DESCRIPTION
**Pull request by SachaZvetelman**
v0.45.0
-------------------------------
*January 27, 2021*

### Changed
- Only call the Checkout `GET` endpoint when the user is logged in.
- Refactored tests.

### CODE REVIEW

File: packages/components/organisms/f-checkout/src/components/Checkout.vue [hunk 1]
```diff
@@ -256,13 +256,17 @@ export default {
...
-            if (!this.isLoggedIn) {
+            if (this.isLoggedIn) {
+                initialLoadPromises.push(this.loadCheckout());
+            } else {
                 await this.loadBasket();
             }
 
-            await Promise.all([this.loadCheckout(), this.loadAvailableFulfilment()]);
+            await Promise.all(initialLoadPromises);
         },
```
Comment (SachaZvetelman) [inline packages/components/organisms/f-checkout/src/components/Checkout.vue]: I personally don't like how this looks in the end. Any thoughts?

Comment (pkedziora) [reply packages/components/organisms/f-checkout/src/components/Checkout.vue]: I prefer Promise.all([this.loadCheckout(), this.loadAvailableFulfilment()]); as it's more readable, does not require to look for the definition of initialLoadPromises

File: packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js [hunk 2]
```diff
@@ -365,6 +350,21 @@ describe('Checkout', () => {
...
+                    const loadCheckoutSpy = jest.spyOn(VueCheckout.methods, 'loadCheckout');
+
+                    shallowMount(VueCheckout, {
+                        store: createStore(),
+                        i18n,
+                        localVue,
+                        propsData
+                    });
+                    await flushPromises();
+
```
Comment (FayeCarter) [inline packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js:368]: Do you need an `// Assert` here?

File: packages/components/organisms/f-checkout/src/components/Checkout.vue (hunk 1 above)
Comment (SachaZvetelman) [reply packages/components/organisms/f-checkout/src/components/Checkout.vue]: But we don't always call `loadCheckout`. That's the method we need to call only if we're authenticated.

Comment (kevinrodrigues) [reply packages/components/organisms/f-checkout/src/components/Checkout.vue]: You could do the below instead? Then you could switch the condition above for the loadBasket() call instead.

```javascript
this.isLoggedIn
? Promise.all([this.loadCheckout(), this.loadAvailableFulfilment()])
: Promise.all([this.loadAvailableFulfilment()])

Comment (SachaZvetelman) [reply packages/components/organisms/f-checkout/src/components/Checkout.vue]: I had pretty much that in my previous commit (https://github.com/justeat/fozzie-components/pull/642/commits/3abfc4bd41dbdb5c941d53f5eabfd44c5ce92583) but didn't really like it either. But I guess it's still better than what I have now.

**Review by FayeCarter — APPROVED** [review]

**Review by kevinrodrigues — APPROVED** [review]

**Pull request merged**